pkg/agent/controller/egress/ip_scheduler.go

```diff
@@ -97,6 +116,60 @@ func NewEgressIPScheduler(cluster memberlist.Interface, egressInformer crdinform
 	return s
 }
 
+func getMaxEgressIPsFromAnnotation(node *corev1.Node) (int, bool, error) {
+	maxEgressIPsStr, exists := node.Annotations[types.NodeMaxEgressIPsAnnotationKey]
+	if !exists {
+		return 0, false, nil
+	}
+	maxEgressIPs, err := strconv.Atoi(maxEgressIPsStr)
+	if err != nil {
+		return 0, false, err
+	}
+	return maxEgressIPs, true, nil
+}
+
+// updateNode processes Node ADD and UPDATE events.
+func (s *egressIPScheduler) updateNode(obj interface{}) {
+	node := obj.(*corev1.Node)
+	maxEgressIPs, found, err := getMaxEgressIPsFromAnnotation(node)
+	if err != nil {
+		klog.ErrorS(err, "The Node's max-egress-ips annotation was invalid", "node", node.Name)
+		if s.deleteMaxEgressIPsByNode(node.Name) {
+			s.queue.Add(workItem)
+		}
+		return
+	}
+	if !found {
+		if s.deleteMaxEgressIPsByNode(node.Name) {
+			s.queue.Add(workItem)
+		}
+		return
+	}
+	if s.updateMaxEgressIPsByNode(node.Name, maxEgressIPs) {
+		s.queue.Add(workItem)
+	}
+}
+
+// deleteNode processes Node DELETE events.
+func (s *egressIPScheduler) deleteNode(obj interface{}) {
+	node, ok := obj.(*corev1.Node)
+	if !ok {
+		deletedState, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			klog.Errorf("Received unexpected object: %v", obj)
+			return
+		}
+		node, ok = deletedState.Obj.(*corev1.Node)
+		if !ok {
+			klog.Errorf("DeletedFinalStateUnknown contains non-Node object: %v", deletedState.Obj)
+			return
+		}
+	}
+	if s.deleteMaxEgressIPsByNode(node.Name) {
+		s.queue.Add(workItem)
```

Comment:
If we need to trigger rescheduling at Node deletion, it should be done even before this commit when there was no per Node annotation?

Reply:
Removed it, there is no need to trigger it because if the Node is selected by any pool, ClusterEventHandler will trigger it.